x509/x509.go

```diff
@@ -1081,8 +1081,8 @@ func Hash(crt *x509.Certificate) string {
 
 // RSACertificateAuthority creates an RSA CA.
 func RSACertificateAuthority(template *x509.Certificate, opts *Options) (ca *CertificateAuthority, err error) {
-	key, e := rsa.GenerateKey(rand.Reader, opts.Bits)
-	if e != nil {
+	key, err := rsa.GenerateKey(rand.Reader, opts.Bits)
+	if err != nil {
 		return
```

Comment:
Previously this code returned `(nil, nil)` if the key can't be generated for some reason.

I would prefer to remove named return arguments in this and other functions – they serve no purpose there and are not used conisistently.

Reply:
I believe this code is wrong anyways, as `err` variable shadows return argument so it returns `(nil, nil)` now

Reply:
The code is right now: `err` is an existing variable, so only `key` is defined on that line, `err` is reused. From https://golang.org/ref/spec#Short_variable_declarations:

> Unlike regular variable declarations, a short variable declaration may redeclare variables provided they were originally declared earlier in the same block (or the parameter lists if the block is the function body) with the same type, **and at least one of the non-blank variables is new**

See also an example in that section.

But I absolutely agree that this code is confusing. I will send another PR with named return parameters removed.

Reply:
ok, nice to know, I was always afraid of such edge cases